### PR TITLE
[POM-415] impelmenting mat-spinner loader, interceptor and service | …

### PR DESCRIPTION
**Pull request by klimtron**
…global loading animation while any http request
do testów polecam włączyć symulację sieci 3g fast lub slow obojętnie:
image

### CODE REVIEW

**Review by pswiderski123 — DISMISSED** [review]

**Review by broslukasz — DISMISSED** [review]

File: src/app/shared/components/loading/loading.component.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+::ng-deep .preloader.mat-progress-spinner circle,
```
Comment (mgol) [inline src/app/shared/components/loading/loading.component.scss]: `::ng-deep` is deprecated, it'd be good to find a better way...

File: src/app/shared/services/loading.service.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    return NEVER.pipe(
+      finalize(() => {
+        this.hide();
+      })
+    );
+  }).pipe(share());
+
+  private show(): void {
+    // Hack avoiding `ExpressionChangedAfterItHasBeenCheckedError` error
+    Promise.resolve(null).then(() => {
```
Comment (mgol) [inline src/app/shared/services/loading.service.ts]: You don't need to pass a value to `Promise.resolve` if you don't use it:
```suggestion
    Promise.resolve().then(() => {
```

File: src/app/shared/components/loading/loading.component.scss (hunk 1 above)
Comment (klimtron) [reply src/app/shared/components/loading/loading.component.scss]: poprawione

File: src/app/shared/services/loading.service.ts (hunk 2 above)
Comment (klimtron) [reply src/app/shared/services/loading.service.ts]: poprawione

**Review by mgol — APPROVED** [review]

**Pull request merged**